### PR TITLE
feat: added clickhouse scripts for sdk events

### PR DESCRIPTION
**Pull request by vsrivatsa-juspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This pull request introduces queries for creating the sdk_event ClickHouse schemas, designed to use Kafka table engines for ingesting events from the vector service. This enhancement aims to improve the efficiency and scalability of event data ingestion by leveraging Kafka's distributed streaming capabilities.

Using Kafka table engines in ClickHouse provides several benefits:

**Real-time Ingestion**: Allows for the efficient, real-time ingestion of event data from the vector service, ensuring up-to-date data availability.
**Scalability**: Kafka's distributed nature handles high throughput and scales easily, accommodating growing event volumes.
**Decoupling**: Decouples data ingestion from processing, leading to more resilient and manageable system architecture.

**Backward Compatibility**:
This change introduces new schemas and tables but does not affect existing schemas or operations. Existing functionalities remain unchanged, ensuring backward compatibility.

### Additional Changes

- [ ] This PR modifies the API contract
- [x] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

**1. Run Docker Compose Command:**

Open your terminal or command prompt.
Navigate to the directory containing your docker-compose.yml file.
Run the following command to start the Kafka UI and ClickHouse server services:
```
docker compose up -d kafka-ui clickhouse-server
```

This command will start the services in detached mode. Make sure that your docker-compose.yml file is correctly configured to include the kafka-ui and clickhouse-server services.

**2. Access ClickHouse Web UI:**

Open your web browser.
Navigate to http://localhost:8123/play.
This URL directs you to the ClickHouse Playground, a web-based interface for interacting with your ClickHouse database.

**3. Run SHOW TABLES query:**

In the ClickHouse Playground, you'll see a SQL query editor.
Enter the following command in the editor:
```
SHOW TABLES;
```

Click the "Run" button or press Ctrl + Enter to execute the command.
This will display a list of tables currently present in your ClickHouse database.

**Expected Outcome**

**Step 1: Docker Compose Command**

The terminal should show that the kafka-ui and clickhouse-server containers are starting up. You can check the status of the services using docker compose ps.

**Step 2: Access ClickHouse Web UI**

The web browser should display the ClickHouse Playground interface at http://localhost:8123/play.

**Step 3: Run SHOW TABLES Command**

The output should list the tables in your ClickHouse database. If you have successfully created the tables as described in the pull request, they should appear in this list.

1. active_payments
2. sdk_active_payments_mv
3. api_active_payments_mv
4. sdk_events
5. sdk_events_audit
6. sdk_events_audit_mv
7. sdk_events_mv
8. sdk_events_parse_errors
9. sdk_events_queue

Screenshot 2024-05-30 at 2 51 13 PM

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible

part of https://github.com/juspay/hyperswitch/issues/4372

### CODE REVIEW

File: crates/analytics/docs/clickhouse/scripts/sdk_events.sql [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    (length(_error) = 0)
+    AND (payment_id IS NOT NULL);
+
+CREATE TABLE sdk_active_payments ( 
+    `payment_id` Nullable(String),
+    `merchant_id` String,
+    `created_at` DateTime,
+    INDEX merchantIndex merchant_id TYPE bloom_filter GRANULARITY 1
+) ENGINE = ReplacingMergeTree
+PARTITION BY toStartOfSecond(created_at)
```
Comment (ivor-juspay) [inline crates/analytics/docs/clickhouse/scripts/sdk_events.sql:213]: is this needed? too many partitions will be created

Comment (vsrivatsa-juspay) [reply crates/analytics/docs/clickhouse/scripts/sdk_events.sql:213]: Yes. There would be 60 active partitions at a given point of time. Rest will be inactive

**Review by ivor-juspay — APPROVED** [review]

File: crates/analytics/docs/clickhouse/scripts/sdk_events.sql [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+CREATE TABLE sdk_active_payments ( 
+    `payment_id` String,
+    `merchant_id` String,
+    `created_at` DateTime64,
+    INDEX merchantIndex merchant_id TYPE bloom_filter GRANULARITY 1
+) ENGINE = MergeTree
+PARTITION BY toStartOfSecond(created_at)
+ORDER BY 
+    (created_at, merchant_id)
```
Comment (vsrivatsa-juspay) [inline crates/analytics/docs/clickhouse/scripts/sdk_events.sql:215]:
```suggestion
    merchant_id
```

File: crates/analytics/docs/clickhouse/scripts/sdk_events.sql [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    `source` LowCardinality(Nullable(String)),
+    `category` LowCardinality(Nullable(String)),
+    `version` LowCardinality(Nullable(String)),
+    `value` Nullable(String),
+    `component` LowCardinality(Nullable(String)),
+    `payment_method` LowCardinality(Nullable(String)),
+    `payment_experience` LowCardinality(Nullable(String)) DEFAULT '',
+    `created_at` DateTime64(3) DEFAULT now64() CODEC(T64, LZ4),
+    `inserted_at`  DateTime DEFAULT now() CODEC(T64, LZ4), 
+    `latency` Nullable(UInt32) DEFAULT 0,
```
Comment (vsrivatsa-juspay) [inline crates/analytics/docs/clickhouse/scripts/sdk_events.sql:131]:
```suggestion
    `latency` Nullable(UInt32) DEFAULT 0
```

**Review by ivor-juspay — DISMISSED** [review]

**Review by lsampras — APPROVED** [review]

**Review by ivor-juspay — APPROVED** [review]

**Review by ivor-juspay — APPROVED** [review]

**Comment by vsrivatsa-juspay** [comment]
moved to #5098

**Pull request closed without merging**